benchmarks/bm_runner.py

```diff
@@ -537,6 +537,66 @@ def func(args: argparse.Namespace) -> None:
         _subprocess_runner([args.asv_sub_command, *args.asv_args], asv=True)
 
 
+class TrialRun(_SubParserGenerator):
+    name = "trialrun"
+    description = (
+        "Fast trial-run a given benchmark, to check it works : "
+        "in a provided or latest-lockfile environment, "
+        "with no repeats for accuracy of measurement."
+    )
+    epilog = (
+        "e.g. python bm_runner.py trialrun "
+        "MyBenchmarks.time_calc /tmp/testpython/bin/python"
+        "\n  NOTE: setting $DATA_GEN_PYTHON is equivalent to the "
+        "'runpath' argument."
+    )
+
+    def add_arguments(self) -> None:
+        self.subparser.add_argument(
+            "benchmark",
+            type=str,
+            help=(
+                "A benchmark name, possibly including wildcards, "
+                "as supported by the ASV '--benchmark' argument."
+            ),
+        )
+        self.subparser.add_argument(
+            "runpath",
+            type=str,
+            nargs="?",
+            help=(
+                "A path to an existing python environment, "
+                "to completely bypass environment building."
+            ),
+        )
+
+    @staticmethod
+    def func(args: argparse.Namespace) -> None:
+        print(args)
+        if args.runpath:
```

Comment:
`pathlib` could `resolve()` `args.runpath` so that relative paths can be passed (my testing indicates that only absolute paths work at the moment).

Reply:
This was trickier than it sounds, because the noxfile sets cwd when running the bm_runner, and bm_runner does the same when running asv.
So, I **_think_** it works to leave the path relative and run bm_runner in the original cwd.
I've done that, and it seems to work.  But I'd be grateful if you'd check it out any way you can think of @trexfeathers

Reply:
I raised this because I tried using a relative path and it couldn't find it, but worked fine when I gave the full path. I don't understand the harm of just fully resolving the path? I had assumed that a fully resolved path would work no matter what the cwd was?

Reply:
Also ab09e95760eae1f0e7060316d9480d54a1220d38 has actioned my suggestion so I'm not sure what we're debating any more

Reply:
> the harm of just fully resolving the path?
You can't resolve the path when running via nox, since nox resets the cwd before calling the bm_runner (and passing the arg).
So if left as it was, a relative path always resolved relative to the ./benchmarks dir instead of the user's cwd.
I did check this + it didn't work !

Reply:
OK I understand now.

Well what you have done works when either `nox` or `python bm_runner.py` is called so that's all good.